### PR TITLE
Add proxy support

### PR DESCRIPTION
**Pull request by mscharp**
Read proxy info from application.properties file first,
then java system settings, and finally from ENV vars.

*Issue #, if available:*
#23, #349 

*Description of changes:*
Merge into the 1.x code line.  However, these changes should be propagated to the 2.x line as well.  

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/main/java/com/amazonaws/services/kinesis/multilang/MultiLangDaemonConfig.java [hunk 1]
```diff
@@ -14,24 +14,18 @@
...
-import org.apache.commons.logging.LogFactory;
-
+import com.amazonaws.ClientConfiguration;
 import com.amazonaws.services.kinesis.clientlibrary.config.KinesisClientLibConfigurator;
 import com.amazonaws.services.kinesis.clientlibrary.lib.worker.KinesisClientLibConfiguration;
 import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import java.io.*;
```
Comment (sahilpalvia) [inline src/main/java/com/amazonaws/services/kinesis/multilang/MultiLangDaemonConfig.java]: Please avoid * imports.

File: src/main/java/com/amazonaws/services/kinesis/multilang/MultiLangDaemonConfig.java [hunk 2]
```diff
@@ -107,6 +106,50 @@ public MultiLangDaemonConfig(String propertiesFile,
         prepare(processingLanguage);
     }
 
+    private ClientConfiguration buildClientConfig(Properties properties)
+    {
```
Comment (sahilpalvia) [inline src/main/java/com/amazonaws/services/kinesis/multilang/MultiLangDaemonConfig.java]: Could you use the [formatter](https://github.com/awslabs/amazon-kinesis-client/tree/master/formatter) provided with the new changes.

**Comment by mlepriberluti** [comment]
@pfifer @mscharp I understand that the merge in the PL https://github.com/awslabs/amazon-kinesis-client/pull/365 has not been done in both the 1.x and 2.x lines.
Is there any supported mode to use the library in conjunction with an HTTP proxy?
Thanks
Mattia

**Comment by deepthakannan** [comment]
Appreciate adding support for proxy configuration. Do you know when this would be available in Maven? https://search.maven.org/search?q=g:com.amazonaws%20AND%20a:amazon-kinesis-client

**Pull request closed without merging**